### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by prius2055**
### Processing data in models

**In this branch I implemented the following;**

- Created model classes for all entities as shown in the ERD diagram
- Set up associations between models.
- Used models to insert data
- Created a custom method that returns the 3 most recent posts for a given user.
- Created a custom method that updates the posts counter for a user.
- Created a custom method that returns the 5 most recent comments for a given post. 
- Created a custom method that updates the comments counter for a post.
- Created a custom method that updates the likes counter for a post.
- Tested all model method in the rails console

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
```
Comment (fobadara) [inline app/models/post.rb]: [REQUIRED] Kindly remove foreign_key has_many :comments and has_many :likes. It is meant to be used when you are trying to reference another model with another name

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,13 @@
...
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
```
Comment (fobadara) [inline app/models/post.rb]: [OPTIONAL] I think your code will be more concise if you had just done this  author.increment!(:posts_counter). You can also include a after_save callback and make the method private this is a good practice.

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,2 +1,8 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_post_comments_counter
+    post.update(comments_counter: post.comments.count)
```
Comment (fobadara) [inline app/models/comment.rb]: [OPTIONAL] I think your code will be more concise if you had just done this  post.increment!(:comments_counter). You can also include a after_save callback and make the method private this is a good practice.

File: app/models/like.rb [hunk 4]
```diff
@@ -1,2 +1,8 @@
 class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_post_likes_counter
+    post.update(likes_counter: post.likes.count)
```
Comment (fobadara) [inline app/models/like.rb]: [OPTIONAL] I think your code will be more concise if you had just done this  post.increment!(:likes_counter). You can also include a after_save callback and make the method private this is a good practice.

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @prius2055, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ CLEAN CODE
✅ SEPARATION OF CONCERNS

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
ers and Happy coding!👏👏👏



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (prius2055) [reply app/models/post.rb]: Done

File: app/models/post.rb (hunk 2 above)
Comment (prius2055) [reply app/models/post.rb]: Done

File: app/models/comment.rb (hunk 3 above)
Comment (prius2055) [reply app/models/comment.rb]: Done

Comment (prius2055) [reply app/models/comment.rb]: Done

**Review by jerryowusu — APPROVED** [review]
Hi @prius2055 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

CookedDelectableDolphin-size_restricted


## Highlights
Made requested changes :+1: 
Neat code :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**